src/openforms/js/components/admin/form_design/logic/actions/Actions.js

```diff
@@ -105,28 +107,62 @@ const ActionFetchFromService = ({action, errors, onChange}) => {
     setIsModalOpen(false);
   };
 
+  const formContext = useContext(FormContext);
+
+  let serviceFetchConfigFromVar =
+    _.cloneDeep(formContext.formVariables.find(element => element.key === action.variable))
+      ?.serviceFetchConfiguration || undefined;
+
+  if (serviceFetchConfigFromVar) {
+    if (!Array.isArray(serviceFetchConfigFromVar.headers)) {
+      serviceFetchConfigFromVar.headers = Object.entries(serviceFetchConfigFromVar.headers || []);
```

Comment:
```suggestion
      serviceFetchConfigFromVar.headers = Object.entries(serviceFetchConfigFromVar.headers || {});
```
this is supposed to convert records into two-tuples, no?

Reply:
This works too because `Object.prototype.isPrototypeOf([]) === true;` :wink:

Reply:
I get why it works, but I don't particularly like relying on quirks like `typeof [] === 'object'`. I'm more concerned about the way this code can be read:

1. default value is an array
2. so this could be read as the implication that `serviceFetchConfigFromVar.headers` is expected to be an array
3. which doesn't hold up with the check above that says it's not an array

this is actually what triggered me to make this remark in the first place - the if-branch looked out of place because if `serviceFetchConfigFromVar.headers` already always is an array, then this would be a dead code branch that never evaluates.

So following the logic where the variable is NOT an array, then what is it expected to be? -> I can only think of an object (excluding arrays and null) and I'd like to see that type consistency in the fallback value too.